spec/bank_account_service_spec.rb

```diff
@@ -0,0 +1,118 @@
+require 'spec_helper'
+require_relative '../services/bank_account_service'
+require_relative '../repositories/bank_account_repository'
+require_relative '../models/bank_account'
+
+RSpec.describe BankAccountService do
+  let(:repo)    { BankAccountRepository.new }
+  let(:service) { BankAccountService.new(repo) }
+
+  before(:each) do
+    BankAccount.dataset.delete
+  end
+
+  describe '#create_account' do
+    it 'creates a new account and returns the BankAccount object' do
+      acct = service.create_account(
+        name:    'Alice',
+        job:     'Engineer',
+        email:   'alice@example.com',
+        address: '123 Main St'
+      )
+      expect(acct).to be_a(BankAccount)
+      expect(acct.name).to eq('Alice')
+      expect(acct.balance).to eq(0.0)
+    end
+
+    it 'raises a uniqueness error when email already exists' do
+      service.create_account(name: 'A', job: 'J', email: 'dup@x.com', address: 'A')
+      result = service.create_account(name: 'B', job: 'K', email: 'dup@x.com', address: 'B')
+      expect(result).to be_a(Hash)
+      expect(result[:error]).to match(/already exists/)
+    end
+  end
+
+  describe '#find_by_email' do
+    it 'returns the account when it exists' do
+      service.create_account(name: 'Bob', job: 'Biz', email: 'bob@x.com', address: 'Addr')
+      acct = service.find_by_email('bob@x.com')
+      expect(acct).to be_a(BankAccount)
+      expect(acct.email).to eq('bob@x.com')
+    end
+
+    it 'returns nil when no account has that email' do
+      expect(service.find_by_email('nope@x.com')).to be_nil
+    end
+  end
+
+  describe '#get_all_accounts' do
+    it 'returns an array of all accounts' do
+      3.times { |i| service.create_account(name: "U#{i}", job: 'X', email: "u#{i}@x.com", address: 'A') }
+      all = service.get_all_accounts
```

Comment:
Use let to declare testing data https://rspec.info/features/3-12/rspec-core/helper-methods/let/